### PR TITLE
Application gateway custom health probe

### PR DESCRIPTION
**Pull request by mytev**

### CODE REVIEW

File: src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Network/ApplicationGateway/Add-ArmApplicationGatewayBackendHttpSettings.ps1 [hunk 1]
```diff
@@ -23,7 +23,9 @@ function Add-ArmApplicationGatewayBackendHttpSettings {
         [int]
         $RequestTimeoutInSec = 30,
         [string]
-        $TrustedRootCertificateName
+        $TrustedRootCertificateName,
+        [PSCustomObject]
```
Comment (mbernard) [inline src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Network/ApplicationGateway/Add-ArmApplicationGatewayBackendHttpSettings.ps1]: Specify Type

File: src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Network/ApplicationGateway/Add-ArmApplicationGatewayHealthProbe.ps1 [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    Param(
+        [PSTypeName("ApplicationGateway")]
+        [Parameter(Mandatory, ValueFromPipeline)]
+        $ApplicationGateway,
+        [ValidatePattern('^(\[.*\]|[a-zA-Z0-9-_.]*)$')]
+        [string]
+        $Name = "default",
+        [string]
+        [ValidateSet("Http", "Https")]
+        $Protocol = "Http",
```
Comment (mbernard) [inline src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Network/ApplicationGateway/Add-ArmApplicationGatewayHealthProbe.ps1]: should be https by default

File: src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Network/ApplicationGateway/Add-ArmApplicationGatewayHealthProbe.ps1 [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        [string]
+        $HostName,
```
Comment (mbernard) [inline src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Network/ApplicationGateway/Add-ArmApplicationGatewayHealthProbe.ps1]: mandatory?
in fact it depends on the PickHostFromBackend.... so we should probably create different ParameterSet for those 2 cases

File: src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Network/ApplicationGateway/Add-ArmApplicationGatewayHealthProbe.ps1 [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        [int]
+        $IntervalInSeconds = 30,
+        [int]
+        $TimeoutInSeconds = 30,
+        [int]
+        $UnhealthyThreshold = 3,
+        [switch]
+        $PickHostNameFromBackendHttpSettings,
+        [int]
+        $MinServers = 0,
```
Comment (mbernard) [inline src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Network/ApplicationGateway/Add-ArmApplicationGatewayHealthProbe.ps1]: Not really obvious, we should find a better name

File: src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Network/ApplicationGateway/Add-ArmApplicationGatewayHealthProbe.ps1 [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        [string]
+        $Path = "/",
+        [int]
+        $IntervalInSeconds = 30,
+        [int]
+        $TimeoutInSeconds = 30,
+        [int]
+        $UnhealthyThreshold = 3,
+        [switch]
+        $PickHostNameFromBackendHttpSettings,
```
Comment (mbernard) [inline src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Network/ApplicationGateway/Add-ArmApplicationGatewayHealthProbe.ps1]: I feel like the default should be true, so we need to reverse that switch

File: src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Network/ApplicationGateway/Add-ArmApplicationGatewayBackendHttpSettings.ps1 [hunk 6]
```diff
@@ -23,7 +23,9 @@ function Add-ArmApplicationGatewayBackendHttpSettings {
...
+        $Probe
```
Comment (mbernard) [inline src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Network/ApplicationGateway/Add-ArmApplicationGatewayBackendHttpSettings.ps1]: Isn't it the probe name here?

**Review by mbernard — CHANGES_REQUESTED** [review]

File: src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Network/ApplicationGateway/Add-ArmApplicationGatewayBackendHttpSettings.ps1 (hunk 6 above)
Comment (mytev) [reply src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Network/ApplicationGateway/Add-ArmApplicationGatewayBackendHttpSettings.ps1]: In the template it looks like 
`"probe" : { "id": "..."}`
The probe name is not enough to get its id. We also need the app gateway `_resourceId`
I can change it to `$ProbeId` instead

File: src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Network/ApplicationGateway/Add-ArmApplicationGatewayHealthProbe.ps1 (hunk 5 above)
Comment (mytev) [reply src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Network/ApplicationGateway/Add-ArmApplicationGatewayHealthProbe.ps1]: I followed Azure portal default values.
I'll change that

**Review by mbernard — APPROVED** [review]

**Pull request merged**